### PR TITLE
TNO-1820: Rework my searches

### PR DESCRIPTION
**Pull request by jdtoombs**
In this PR: 

- rework filter to send raw json for elastic search query
- add ability to save the filter similar to the editor
- display saved filters in a list
- edit said filters so they can be later used in reports

my-searches

### CODE REVIEW

**Review by Fosol — CHANGES_REQUESTED** [review]

File: libs/net/models/Areas/Subscriber/Models/Report/FilterModel.cs [hunk 1]
```diff
@@ -23,7 +29,7 @@ public class FilterModel : BaseTypeWithAuditColumnsModel<int>
     /// <summary>
     /// get/set - The filter settings.
     /// </summary>
-    public JsonDocument Settings { get; set; } = JsonDocument.Parse("{}");
+    public FilterSettingsModel Settings { get; set; } = new();
```
[the remark this answers is not in the document]
Comment (jdtoombs) [reply libs/net/models/Areas/Subscriber/Models/Report/FilterModel.cs:32]: Previously it wasn't keeping all the values when I was adding/updating so I made it the same as we did on the editor side and it began saving the other setting values!

File: app/subscriber/src/features/my-searches/utils/useLookupOptions.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import { ILookupController, useLookup } from 'store/hooks';
+import { ILookupState } from 'store/slices';
+import { getSortableOptions, getSourceOptions, IOptionItem } from 'tno-core';
+
+export interface ILookupOptionsState extends ILookupState {
```
[the remark this answers is not in the document]
Comment (jdtoombs) [reply app/subscriber/src/features/my-searches/utils/useLookupOptions.ts]: moved to more generic location

File: api/net/Areas/Subscriber/Controllers/FilterController.cs [hunk 3]
```diff
@@ -142,7 +154,7 @@ public IActionResult Delete(FilterModel model)
         var user = _userService.FindByUsername(username) ?? throw new NotAuthorizedException("User does not exist");
         var filter = _filterService.FindById(model.Id) ?? throw new NoContentException("Filter does not exist");
         if (filter.OwnerId != user?.Id) throw new NotAuthorizedException("Not authorized to delete filter");
-        _filterService.DeleteAndSave(filter);
+        _filterService.DeleteAndSave(model.ToEntity(_serializerOptions));
```
[the remark this answers is not in the document]
Comment (jdtoombs) [reply api/net/Areas/Subscriber/Controllers/FilterController.cs]: copy and paster error, adjusted.

File: libs/net/models/Areas/Subscriber/Models/Report/FilterModel.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Fosol) [reply libs/net/models/Areas/Subscriber/Models/Report/FilterModel.cs:32]: That doesn't make sense to me.  @kylgarmor did you recently run into this issue?

**Review by Fosol — APPROVED** [review]

**Pull request merged**